### DISCUSSION TITLE
How to Write File on Vercel ?

### DISCUSSION THREAD

**Discussion by m0x61h0x64i** [vercel/community#794] [Help] [answered]
Hi, i wanted to check if vercel deployed my application as well so i managed to get the log of mongoose database connection data and Write them to a file called ok.html and ok.txt but its only working locally and doesnt work on vercel. i created a public directory and linked it to my index of my project and i had the following code to write mongoose connection data to a file but it was not successful. any helps please ?

```bash
const mongoose = require('mongoose').connect
const fs = require('fs')

mongoose(process.env.MONGODB_URL, (e) => {
    if (e) {
        return fs.writeFileSync('public/ok.html', `it doesnt work: ${e}`)
    }

    fs.writeFileSync('public/ok.html', 'it works')
})
```

**Comment by amyegan** [comment, marked as answer]
Writing files from a Serverless Function is limited to the `/tmp` directory. You can read more about recommended alternatives here:
- https://github.com/vercel/community/discussions/241
- https://github.com/vercel/vercel/discussions/5320#discussioncomment-110775
- https://github.com/vercel/vercel/discussions/5190#discussioncomment-79190
